collections/_posts/2022-11-28-CurriculumLearningByDIH.md

```diff
@@ -0,0 +1,117 @@
+
+# Methods
+
+## Dynamic Instantaneous Hardness
+
+The goal of the measure is to get the usefulness of each sample in the future in order to select them wisely. The Dynamic Instantaneous Hardness (DIH) is defined as the exponential moving average of instantaneous hardness measures of a sample over time, the instantaneous hardness $$a_t(i)$$ being a measure retrieved from the epoch that has just been computed.
+
+$$
+r_{t+1}(i) = \left\{
+  \begin{array}
+    1 \gamma*a_t(i) + (1-\gamma)*r_{t}(i) \quad if \quad i \in S_t \\
+    r_t(i) \quad else
+  \end{array}
+  \right.
+$$
+
+$$ \gamma \in [0,1] $$ is a discount factor and $$S_t$$ is the set of samples used for training at time $$t$$.
+
+The three instantaneous hardness used to calculate DIH and compared are:
+* the loss : $$l(y_i,F(x_i; w_t))$$, where $$l(·, ·)$$ is a standard loss function and $$F(·; w)$$ is the model where $$w$$ are the model parameters
+* the loss change between two consecutive time steps : $$ \|l(y_i,F(x_i; w_t))-l(y_i,F(x_i; w_{t-1}))\| $$
+* the prediction flip (the 0-1 indicator of whether the prediction correctness changes) between two consecutive time steps : $$ \| 𝟙[\hat{y}^t_i = y_i]- 𝟙[\hat{y}^{t-1}_i = y_i]\| $$, where $$\hat{y}^t_i$$ is the prediction of sample $$i$$ in step $$t$$, e.g., $$ \arg\max_j F(x_i; w_t)[j]$$ for classification.
+
+![](/collections/images/CL_DIH/DIH_vs_SPL.jpg)
+
+Compared to DIH, instantaneous hardness is i) much less stable during training and ii) requires extra training and inference steps by a model over all the samples.
+
+
+## Properties
+
+DIH can vary a lot between different samples facilitating their discrimination, and so helping to select the most informative samples. When it is smaller, the samples are more memorable, i.e. easy to learn, while a larger DIH means that the samples are harder to retain.
+
+![](/collections/images/CL_DIH/sample_separation.jpg)
+The the large steps in the curves are explained by a cyclic learning rate. We clearly see the gap between the training dynamics of red samples that are harder and blue samples that have smaller DIH. Moreover, the variance of samples with larger DIH is bigger showing that a local minima is not found for them. It is thus necessary to revisit them often, and the same conclusion can be made regarding the prediction flip that is higher for the same samples.
```

Comment:
Repeated word + the last sentence is still confusing.
> the same conclusion can be made regarding the prediction flip that is higher for the same samples

does not really make sense. Do you mean to say that the prediction flip is just as useful as the DIH to help discriminate which samples to revisit? In that case, _regarding_ is not the right word because it's not indicating is a justification for why the same conclusion can be made.
```suggestion
The large steps in the curves are explained by a cyclic learning rate. We clearly see the gap between the training dynamics of red samples that are harder and blue samples that have smaller DIH. Moreover, the variance of samples with larger DIH is bigger showing that a local minima is not found for them. It is thus necessary to revisit them often, and the same conclusion can be made regarding the prediction flip that is higher for the same samples.
```

Reply:
Yes that's almost it: i want to say that the prediction flip, as the loss variance, is showing that the samples with large DIH are those that we we need to revisit. I changed the order of the sentence, trying to make it more clear

Reply:
Okay, I think I finally get it with the last version! Thanks for your patience with the numerous edits :slightly_smiling_face: